include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.h

```diff
@@ -0,0 +1,242 @@
+  get_mapping() const;
+
+  dealii::FiniteElement<dim> const &
+  get_fe_p() const;
+
+  dealii::FiniteElement<dim> const &
+  get_fe_u() const;
+
+  dealii::DoFHandler<dim> const &
+  get_dof_handler_p() const;
+
+  dealii::DoFHandler<dim> const &
+  get_dof_handler_u() const;
+
+  dealii::AffineConstraints<Number> const &
+  get_constraint_p() const;
+
+  dealii::AffineConstraints<Number> const &
+  get_constraint_u() const;
+
+  dealii::types::global_dof_index
+  get_number_of_dofs() const;
+
+  /*
+   * Initialization of vectors.
+   */
+  void
+  initialize_dof_vector(BlockVectorType & dst) const final;
+
+  /*
+   * Prescribe initial conditions using a specified analytical/initial solution function.
+   */
+  void
+  prescribe_initial_conditions(BlockVectorType & dst, double const time) const final;
+
+  /*
+   *  This function is used in case of explicit time integration:
+   *  This function evaluates the right-hand side operator, the
+   *  convective and viscous terms (subsequently multiplied by -1.0 in order
+   *  to shift these terms to the right-hand side of the equations)
+   *  and finally applies the inverse mass operator.
+   */
+  void
+  evaluate(BlockVectorType & dst, BlockVectorType const & src, double const time) const final;
+
+  /*
+   * Operators.
+   */
+
+  // acoustic operator
+  void
+  evaluate_acoustic_operator(BlockVectorType &       dst,
+                             BlockVectorType const & src,
+                             double const            time) const;
+
+  // inverse velocity mass operator
+  void
+  apply_inverse_mass_operator(BlockVectorType & dst, BlockVectorType const & src) const;
+
+private:
+  void
+  initialize_dof_handler_and_constraints();
+
+  void
+  initialize_operators();
+
+  /*
+   * Grid
+   */
+  std::shared_ptr<Grid<dim> const> grid;
+
+  /*
+   * dealii::Mapping (In case of moving meshes (ALE), this is the dynamic mapping describing the
+   * deformed configuration.)
+   */
+  std::shared_ptr<dealii::Mapping<dim> const> mapping;
+
+  /*
+   * User interface: Boundary conditions and field functions.
+   */
+  std::shared_ptr<BoundaryDescriptor<dim> const> boundary_descriptor;
+  std::shared_ptr<FieldFunctions<dim> const>     field_functions;
+
+  /*
+   * List of parameters.
+   */
+  Parameters const & param;
+
+  /*
+   * A name describing the field being solved.
+   */
+  std::string const field;
+
+  /*
+   * Basic finite element ingredients.
+   */
+  std::shared_ptr<dealii::FiniteElement<dim>> fe_p;
+  std::shared_ptr<dealii::FiniteElement<dim>> fe_u;
+
+  dealii::DoFHandler<dim> dof_handler_p;
+  dealii::DoFHandler<dim> dof_handler_u;
+
+  dealii::AffineConstraints<Number> constraint_p, constraint_u;
+
+  std::string const dof_index_p = "pressure";
+  std::string const dof_index_u = "velocity";
+
+  std::string const quad_index_p = "pressure";
+  std::string const quad_index_u = "velocity";
+  // quadrature that works for both, pressure and velocity
```

Comment:
could you explain what this means or for which purpose this quadrature is needed/used?

Reply:
This might be confusing with the present PR. It is possible to use different orders for pressure and velocity, since we have a merged operator we need a quadrature rule that works for both orders, i.e. `n_q=(k_max+1)^dim`.

Reply:
So `quad_index_p/u` is only used for the mass matrix problem?

Reply:
Currently, yes. However, I plan to add convective terms where `quad_index_p/u` will also be needed.

Reply:
I added a comment in [3cdb937](https://github.com/exadg/exadg/pull/592/commits/3cdb93776eac0fb46fe68be7691829cbc5e32e75)